test_jpeg/jpegenc/subblocks/RLE/RLECore/rlecore.py

```diff
@@ -0,0 +1,179 @@
+""" core of the run length encoder module """
+
+from myhdl import always_comb, always_seq, block
+from myhdl import intbv, ResetSignal, Signal
+from myhdl.conversion import analyze
+from jpegenc.subblocks.RLE.RLECore.entropycoder import entropycoder
+
+
+WIDTH_RAM_ADDRESS = 6
+WIDTH_RAM_DATA = 12
+SIZE = 4
+
+class DataStream(object):
```

Comment:
Eventually, the interfaces need to be placed in an interface python module or something, but that can be deferred till later.